js/parser.js

```diff
@@ -191,11 +196,181 @@ var ERMrest = (function(module) {
                     }
                 }
 
-                context.filter = new ParsedFilter(type);
-                context.filter.setFilters(filters);
+                this._firstFilter = new ParsedFilter(type);
+                this._firstFilter.setFilters(filters);
+            }
+        }
+
+    }
+
+    Location.prototype = {
+
+        get uri() {
+            return this._uri;
+        },
+
+        get compactUri() {
+            return this._compactUri;
+        },
+
+        get service() {
+            return this._service;
+        },
+
+        get catalog() {
+            return this._catalog;
+        },
+
+        get api() {
+            return this._api;
+        },
+
+        get path() {
+            return this._path;
+        },
+
+        get compactPath() {
+            return this._compactPath;
+        },
+
+        get firstSchemaName() {
```

Comment:
Technically, its not really "first" schema name or table name. The real semantics here are that these are what in ERMrest-speak we sometimes call the "context" of the URI. Meaning, the thing being operated on (either updated or retrieved or deleted).

A more conventional way of saying that _might_ be to say this is the "projection" of the URL. Meaning that you have some URL `entity/s:t/c=123` and in an "entity" interface you are "projecting" all of the columns of the table "`s:t`".

This isn't a _blocker_ and we need to give this more thought in order to adequately resolve the issue. For the purpose of this PR, I suggest just putting a note in the jsdoc string indicating that these two properties are subject to change soon.